lib/helpers/dart_sytax_highlighter.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart' show Color, TextSpan, TextStyle, Theme;
+import 'package:flutter_markdown/flutter_markdown.dart';
+import 'package:highlight/highlight.dart';
+
+class DartSyntaxHighlighter extends SyntaxHighlighter {
+  DartSyntaxHighlighter(this.context);
+
+  final String _rootKey = 'root';
+  final BuildContext context;
+
+  @override
+  TextSpan format(String source) {
+    return TextSpan(
+      style: _syntaxTheme[_rootKey],
+      children: _convert(
+        highlight.parse(source, language: 'dart').nodes!,
+      ),
+    );
+  }
+
+  List<TextSpan> _convert(List<Node> nodes) {
+    final List<TextSpan> spans = [];
+    var currentSpans = spans;
```

Comment:
Yes, but in this case item added to `currentSpans` are present in `spans` too.
Creating and assigning a new list won't work (_result is empty_) because the lists need to be connected.
But it's not needed actually. I'll get rid of `spans` and only keep `currentSpans`, as there's the same result as before.